flume-ng-sinks/flume-ng-kafka-sink/src/main/java/org/apache/flume/sink/kafka/KafkaSink.java

```diff
@@ -437,7 +438,7 @@ private void setProducerProps(Context context, String bootStrapServers) {
   private Headers toKafkaHeaders(Map<String, String> headers) {
     Headers kafkaHeaders = new RecordHeaders();
     for (Entry<String, String> header : headers.entrySet()) {
-      kafkaHeaders.add(header.getKey(), header.getValue().getBytes());
+      kafkaHeaders.add(header.getKey(), header.getValue().getBytes(Charsets.UTF_8));
```

Comment:
I think expected Charsets should should be configurable through flume context parameters

Reply:
+1